docs/design/proposals/kms-custom-http.md

```diff
@@ -0,0 +1,40 @@
+# External Storage of Encryption Passphrases
+
+## Goal
+
+The current KMS methods are modeled to provide an end-to-end solution for managing the encryption passphrases that are used by ceph-csi. While they are fine and use top-notch services like Vault and Amazon KMS, they do struggle to fit into configurations where the encryption is handleded completely by a dedicated in-house service.
+
+The goal is to allow configuring a very thin kms method that completely externalizes the storage and retrieval of passphrases to an opaque service.
+
+## Solution Requirements
+
+1. For new volumes the generated passphrases should be sent to the service for persisting.
+1. For existing volumes the passphrases should be fetched from the aftermentioned service.
+1. The communication should be secure because the passphrases are transferred in plain format.
+
+## Suggested Implementation
+
+New KMS type: `opaqueservice`
+
+Ultimately, we strongly suggest using mTLS for communicating with the endpoint. However, the security level is up to the system operator to decide so we are better off allowing several options.
+
+Example configuration:
+```json
+{
+  "mykms": {
+    "encryptionKMSType": "opaqueservice",
+    "serviceurl": "https://my.service.com/tokens",
+    "mtls": "true",
+    "basicauth": "false"
```

Comment:
In case you would enable `basicauth`, where/how would a user configure login details?

Reply:
rather curious to know how its going to make a decision/logic  behind `enabled basicauth`?

Reply:
I assume it is https://datatracker.ietf.org/doc/html/rfc7617 and needs options for usersname/password

Reply:
I assume like aws kms, in a separate secret. I would to define the behavior in the config and not guess based on runtime basic auth challenge responses.

Reply:
>I assume like aws kms, in a separate secret. I would prefer to define the behavior in the config and not guess based on runtime basic auth challenge responses.

@tareqhs Adding an example of such a secret/config in design doc would help explain this better. 

You may also be interested in selecting different such secret/configs according to kubernetes namespace / tenant where PVC was created like in other encryptions currently available (just a suggestion).